pkg/asset/agent/agentconfig/agent_config.go

```diff
@@ -34,9 +40,49 @@ func (*Asset) Dependencies() []asset.Asset {
 	return []asset.Asset{}
 }
 
-// Generate generates the Agent manifest.
+// Generate generates the Agent Config manifest.
 func (a *Asset) Generate(dependencies asset.Parents) error {
-	return nil
+	agentConfig := &agent.Config{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "AgentConfig",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "example-agent-config",
+			Namespace: "cluster0",
+		},
+		Spec: agent.Spec{
+			RendezvousIP: "your-node0-ip",
+			Hosts: []agent.Host{
+				{
+					Hostname: "change-to-hostname",
+					Role:     "master",
+					RootDeviceHints: baremetal.RootDeviceHints{
+						DeviceName: "/dev/sda",
+					},
+					Interfaces: []*aiv1beta1.Interface{
+						{
+							Name:       "your-network-interface-name",
+							MacAddress: "00:00:00:00:00",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	a.Config = agentConfig
```

Comment:
We probably shouldn't do this. We want to create the file if nothing else is consuming this (i.e. if running `agent create agent-config`), but we don't actually want to pass the contents of the default file on to other assets in the graph.

We'd still have to detect the defaults in `Load()` as well, so that if the user does `agent create agent-config` and then does not customise the file, future steps (like `agent create cluster-manifests`/`image`) won't fail. Or would we?? Maybe it's OK to say that if you explicitly ask for the template for that file, it's an error if you don't replace the bogus values with real ones?

Reply:
Should we make it more obvious that "openshift agent create agent-config" is currently generating a template? Perhaps name it "agent-config-template"? And then change it when the interactive survey is ready.

I think it would be ok if the user realized that they generated a template, that later on when they run "create cluster-manifests" that they may see errors.